go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -655,16 +673,36 @@ func (c *Controller) updateEIP(existing *state, update *config) error {
 			return fmt.Errorf("failed to delete egress IP address %s from annotation: %v", existing.eIPConfig.addr.String(), err)
 		}
 	}
-	if (update == nil && existing.eIPConfig != nil && existing.eIPConfig.route != nil) ||
-		(update != nil && update.eIPConfig != nil && update.eIPConfig.route != nil &&
-			existing.eIPConfig != nil && existing.eIPConfig.route != nil &&
-			!existing.eIPConfig.route.Equal(*update.eIPConfig.route)) {
-		// route manager takes care of retry
-		c.routeManager.Del(*existing.eIPConfig.route)
+	// delete stale routes
+	// existing routes need to be deleted if there's no update and if there's no other active egress IP on this link.
+	if update == nil && existing.eIPConfig != nil && len(existing.eIPConfig.routes) > 0 && existing.eIPConfig.addr != nil {
+		// Egress IP for this config and link should already be deleted in steps previously.
+		// If there is different Egress IP active on this link, we do not want to delete the routes needed for that other egress IP.
+		ipFamily := getIPFamily(utilnet.IsIPv6(existing.eIPConfig.addr.IP))
+		assignedAddresses, err := c.getAnnotation()
+		if err != nil {
+			return fmt.Errorf("failed to get assigned addresses: %v", err)
+		}
+		isEIPOnLink, err := isEgressIPOnLink(existing.eIPConfig.addr.LinkIndex, ipFamily, assignedAddresses)
+		if err != nil {
+			return fmt.Errorf("failed to determine if link with index %d hosts an existing Egress IP: %v",
+				existing.eIPConfig.addr.LinkIndex, err)
+		}
+		if !isEIPOnLink {
```

Comment:
since above you mention egressIP should have been removed on the link already, if isEIPOnLink we have a bug? Should we have an else statement logging an error?

Reply:
no, its possible that there are multiple EIPs assigned to the same link. It wouldnt be a bug.